src/pudl_archiver/archivers/epacems.py

```diff
@@ -34,29 +71,31 @@ async def get_resources(self) -> ArchiveAwaitable:
         if file_list.status_code == 200:
             resjson = file_list.content.decode("utf8").replace("'", '"')
             file_list.close()  # Close connection.
-            bulk_files = json.loads(resjson)
+            bulk_files = (BulkFile(**f) for f in json.loads(resjson))
             quarterly_emissions_files = [
                 file
                 for file in bulk_files
-                if (file["metadata"]["dataType"] == "Emissions")
-                and (file["metadata"]["dataSubType"] == "Hourly")
-                and ("quarter" in file["metadata"])
-                and (self.valid_year(file["metadata"]["year"]))
+                if (file.metadata.data_type == "Emissions")
+                and (file.metadata.data_sub_type == "Hourly")
+                and (file.metadata.quarter in {1, 2, 3, 4})
+                and self.valid_year(file.metadata.year)
             ]
             logger.info(f"Downloading {len(quarterly_emissions_files)} total files.")
             logger.debug(f"File info: {quarterly_emissions_files}")
-            for i, cems_file in enumerate(quarterly_emissions_files):
-                yield self.get_year_quarter_resource(
-                    file=cems_file, request_count=i + 1
-                )  # Count files starting at 1 for human legibility.
+            files_by_year = groupby(
+                sorted(quarterly_emissions_files, key=lambda bf: bf.metadata.year or 0),
```

Comment:
Why 0 here?

Reply:
This is also type stuff to handle "year might not be in metadata." But I think there's a way around it :)